File: matiasmasso/Mat.NET/DT2/DTOBoxItem.vb
Public Class DTOBoxItem
    Public Property Title As String
    Public Property Footer As String
    Public Property ImageUrl As String
    Public Property NavigateUrl As String

    Public Property Tag As DTOBaseGuid


    Shared Function Html(oBoxItem As DTOBoxItem) As String
        Dim sb As New System.Text.StringBuilder
        sb.AppendLine(vbTab & "<div class='BoxItemContainer'>")
        sb.AppendLine(vbTab & vbTab & "<div class='BoxItemTitle'>" & oBoxItem.Title & "</div>")
        sb.AppendLine(vbTab & vbTab & "<div class='BoxItemImgContainer' >")
        sb.AppendLine(vbTab & vbTab & vbTab & "<a href='" & oBoxItem.NavigateUrl & "'>")
        sb.AppendLine(vbTab & vbTab & vbTab & vbTab & "<img src='" & oBoxItem.ImageUrl & "'/>")
        sb.AppendLine(vbTab & vbTab & vbTab & "</a>")
        sb.AppendLine(vbTab & vbTab & "</div>")
        sb.AppendLine(vbTab & vbTab & "<div class='BoxItemFooter'>" & oBoxItem.Footer & "</div>")
        sb.AppendLine(vbTab & "</div>")
        Dim retval As String = sb.ToString
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOBoxItem.cs
namespace DTO
{
    public class DTOBoxItem
    {
        public string Title { get; set; }
        public string Footer { get; set; }
        public string ImageUrl { get; set; }
        public string NavigateUrl { get; set; }

        public DTOBaseGuid Tag { get; set; }


        public static string Html(DTOBoxItem oBoxItem)
        {
            System.Text.StringBuilder sb = new System.Text.StringBuilder();
            sb.AppendLine("\t" + "<div class='BoxItemContainer'>");
            sb.AppendLine("\t" + "\t" + "<div class='BoxItemTitle'>" + oBoxItem.Title + "</div>");
            sb.AppendLine("\t" + "\t" + "<div class='BoxItemImgContainer' >");
            sb.AppendLine("\t" + "\t" + "\t" + "<a href='" + oBoxItem.NavigateUrl + "'>");
            sb.AppendLine("\t" + "\t" + "\t" + "\t" + "<img src='" + oBoxItem.ImageUrl + "'/>");
            sb.AppendLine("\t" + "\t" + "\t" + "</a>");
            sb.AppendLine("\t" + "\t" + "</div>");
            sb.AppendLine("\t" + "\t" + "<div class='BoxItemFooter'>" + oBoxItem.Footer + "</div>");
            sb.AppendLine("\t" + "</div>");
            string retval = sb.ToString();
            return retval;
        }
    }
}
